server/src/main/java/com/thoughtworks/go/server/service/ServerConfigService.java

```diff
@@ -173,4 +179,20 @@ public SiteUrls getServerSiteUrls() {
     public void createOrUpdateServerSiteUrls(SiteUrls siteUrls) {
         goConfigService.updateConfig(new CreateOrUpdateConfigServerSiteUrlsCommand(siteUrls), currentUsername());
     }
+
+    public ArtifactConfig getArtifactsConfig() {
+        return serverConfig().getArtifactConfig();
+    }
+
+    public void updateArtifactConfig(ArtifactConfig modifiedArtifactConfig, HttpLocalizedOperationResult operationResult) {
```

Comment:
We are trying to move away from the pattern of sending `HttpLocalizedOperationResult` or any such class to the service. Instead trying throwing appropriate exceptions in service which the controller catches and renders the result